crates/agglayer-storage/src/error.rs

```diff
@@ -0,0 +1,11 @@
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Trying to access an unknown ColumnFamily")]
+    ColumnFamilyNotFound,
+
+    #[error("Serialization error: {0}")]
+    Serialization(#[from] bincode::Error),
+    #[error(r#"An unexpected error occurred: {0}
+        This is a critical bug that needs to be reported on `https://github.com/agglayer/agglayer/issues`"#)]
+    Unexpected(String),
```

Comment:
Would a `Serialization` issue also qualify as a critical bug that needs to be reported? I'd expect the data structures that are stored in the permanent storage to be defined in a way to make them always serializable without an issue.

Reply:
Yes we can do that